core/src/main/scala/org/http4s/Credentials.scala

```diff
@@ -21,75 +21,75 @@ package org.http4s
 import java.nio.charset.StandardCharsets
 import java.util.Base64
 
-import org.http4s.util.{Renderable, Writer}
+import org.http4s.util.{NonEmptyList, Renderable, Writer}
 
 sealed abstract class Credentials extends Renderable {
   def authScheme: AuthScheme
-  def value: String
 }
 
-final case class BasicCredentials(username: String, password: String) extends Credentials {
-  val authScheme = AuthScheme.Basic
+object Credentials {
+  final case class Token(authScheme: AuthScheme, token: String)
+      extends Credentials {
+    def render(writer: Writer): writer.type =
+      writer << authScheme << ' ' << token
+  }
 
-  override lazy val value = {
+  final case class AuthParams(authScheme: AuthScheme, params: NonEmptyList[(String, String)])
+      extends Credentials {
+    def render(writer: Writer): writer.type = {
+      writer << authScheme
+      writer << ' '
+      var first = true
+      params.foreach { case (k, v) =>
+        if (first) first = false
+        else writer.append(',')
+
+        if (k.isEmpty) writer << '"'
+        else writer<< k << '=' << '"'
+
+        v.foreach {
+          case '"' => writer << '\\' << '"'
+          case '\\' => writer << '\\' << '\\'
+          case c => writer << c
+        }
+        writer << '"'
+      }
+      writer
+    }
+  }
+
+  object AuthParams {
+    def apply(authScheme: AuthScheme, param: (String, String), params: (String, String)*): AuthParams =
+      apply(authScheme, NonEmptyList(param, params:_*))
+  }
+}
+
+final case class BasicCredentials(username: String, password: String) {
+  lazy val token = {
     val userPass = username + ':' + password
     val bytes = userPass.getBytes(StandardCharsets.ISO_8859_1)
-    val cookie = Base64.getEncoder.encodeToString(bytes)
-    "Basic " + cookie
+    Base64.getEncoder.encodeToString(bytes)
```

Comment:
This function now just returns the token, not the entire value of the Authorization header.

```
scala> headers.Authorization(BasicCredentials("admin", "password"))
res7: org.http4s.headers.Authorization = Authorization: Basic YWRtaW46cGFzc3dvcmQ=
```